crates/core_arch/src/x86/avx512f.rs

```diff
@@ -8704,6 +8704,21 @@ pub unsafe fn _mm512_maskz_compress_epi32(k: __mmask16, a: __m512i) -> __m512i {
     ))
 }
 
+/// Contiguously store the active 32-bit integers in a (those with their respective bit set in writemask k) to unaligned memory at base_addr.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=512_mask_compressstoreu_epi32&expand=1229)
+#[inline]
+#[target_feature(enable = "avx512f")]
+#[cfg_attr(test, assert_instr(vpcompressd))]
+pub unsafe fn _mm512_mask_compressstoreu_epi32(mem_addr: *mut i32, k: __mmask16, a: __m512i) {
+    let r = transmute(vpcompressd(
+        a.as_i32x16(),
+        _mm512_setzero_si512().as_i32x16(),
+        k,
+    ));
+    ptr::write_unaligned(mem_addr as *mut __m512i, r);
```

Comment:
This is incorrect: the intrinsic is only supposed to store the elements selected by the mask, not the full 512 bits. This is done with a special LLVM intrinsic: https://godbolt.org/z/vWf7KE

Reply:
The memory and register forms of the instruction behave differently: https://www.felixcloutier.com/x86/vcompressps

> Memory destination version: Only the contiguous vector is written to the destination memory location. EVEX.z must be zero.

> Register destination version: If the vector length of the contiguous vector is less than that of the input vector in the source operand, the upper bits of the destination register are unmodified if EVEX.z is not set, otherwise the upper bits are zeroed.

Reply:
I think this will require special support in the compiler. We don't currently have a way to express vectors of `i1`.